rest/server_context.go

```diff
@@ -264,16 +275,18 @@ func (sc *ServerContext) PostUpgrade(preview bool) (postUpgradeResults PostUpgra
 	return postUpgradeResults, nil
 }
 
+// Removes and re-adds a database to the ServerContext.
+func (sc *ServerContext) _reloadDatabaseFromConfig(reloadDbName string) (*db.DatabaseContext, error) {
+	sc._removeDatabase(reloadDbName)
+	config := sc.dbConfigs[reloadDbName]
+	return sc._getOrAddDatabaseFromConfig(*config, true)
```

Comment:
If `_getOrAddDatabaseFromConfig` returns error, we'll end up with mismatched sets in `sc.databases_` and `sc.dbConfigs`.  Is that ok?  I don't see anywhere we're currently blocking if a config already exists in dbConfigs- we're just overwriting in all cases, I think - but wanted to confirm.

Reply:
This is an area that will need a bit more attention in the PRs that add the `/db/_config` REST APIs and background config polling to ensure things are consistent and safe.